main.go

```diff
@@ -79,72 +80,72 @@ func ndkHome() string {
 	return "ndk-bundle"
 }
 
-func inPath(path string) bool {
-	return strings.Contains(os.Getenv("PATH"), path)
+func newNDKHome() (string, error) {
+	if v := os.Getenv("ANDROID_HOME"); v != "" {
+		// $ANDROID_HOME is deprecated
+		return filepath.Join(v, "ndk-bundle"), nil
+	}
+	if v := os.Getenv("ANDROID_SDK_ROOT"); v != "" {
+		// $ANDROID_SDK_ROOT is preferred over $ANDROID_HOME
+		return filepath.Join(v, "ndk-bundle"), nil
+	}
+	return "", errors.New("neither $ANDROID_HOME nor $ANDROID_SDK_ROOT is specified")
 }
 
 func updateNDK(revision string) error {
-	ndkURL := ndkDownloadURL(revision)
-	ndkHome := ndkHome()
+	ndkDownloadURL := ndkDownloadURL(revision)
+	currentNdkHome := currentNDKHome()
 
-	currentRevision := installedNDKVersion(ndkHome)
+	currentRevision := ndkRevision(currentNdkHome)
 	if currentRevision == revision {
-		log.Donef("NDK r%s already installed at %s", revision, ndkHome)
+		log.Donef("NDK r%s already installed at %s", revision, currentNdkHome)
 		return nil
 	}
 
 	if currentRevision != "" {
-		log.Printf("NDK version %s found at: %s", currentRevision, ndkHome)
+		log.Printf("NDK version %s found at: %s", currentRevision, currentNdkHome)
 	}
 
 	log.Printf("Removing NDK...")
-	if err := os.RemoveAll(ndkHome); err != nil {
+	if err := os.RemoveAll(currentNdkHome); err != nil {
 		return err
 	}
 	log.Printf("Done")
 
 	log.Printf("Downloading NDK r%s...", revision)
-	// The NDK archive contents are wrapped in an extra subdirectory, so we unzip to the parent directory,
-	// then rename the subdirectory to ndk-bundle
-	subDirPattern, err := regexp.Compile("android-ndk-.*")
+	newNDKHome, err := newNDKHome()
 	if err != nil {
 		return err
 	}
-
-	unzipTarget := filepath.Dir(ndkHome)
-	if err := pathutil.EnsureDirExist(unzipTarget); err != nil {
+	newNDKHomeParentDir := filepath.Dir(newNDKHome)
+	if err := pathutil.EnsureDirExist(newNDKHomeParentDir); err != nil {
 		return err
 	}
-	if err := command.DownloadAndUnZIP(ndkURL, unzipTarget); err != nil {
+	if err := command.DownloadAndUnZIP(ndkDownloadURL, newNDKHomeParentDir); err != nil {
 		return err
 	}
 
-	var unzippedDirName string
-	dirs, err := ioutil.ReadDir(unzipTarget)
-	for _, dir := range dirs {
-		if subDirPattern.MatchString(dir.Name()) {
-			unzippedDirName = dir.Name()
-		}
+	// The NDK archive contents are wrapped in an extra subdirectory, so we unzip to the parent directory,
+	// then rename the subdirectory to ndk-bundle
```

Comment:
This comment rather belongs above the `DownloadAndUnZIP` call I think because we use `newNDKHomeParentDir` there.